core/object.go

```diff
@@ -49,6 +49,9 @@ const (
 	TagObject      ObjectType = 4
 	OFSDeltaObject ObjectType = 6
 	REFDeltaObject ObjectType = 7
+
+	// Used as a placeholder to request an object of any time.
```

Comment:
I'm thinking it will be nice to explain that all these types and their associated numbers are from the official git specification, except for InvalidObject and AnyObject, which are object types of our own invention, *that cannot be found inside a public repository*.